scripts/tuple_tree_generator/tuple_tree_generator/schema/struct.py

```diff
@@ -136,19 +143,18 @@ def __init__(self, *, name, pointee_type, root_type, doc=None, optional=False, c
     def type(self):  # noqa: A003
         return f"TupleTreeReference<{self.pointee_type}, {self.root_type}>"
 
-    @property
-    def python_type(self):
+    def type_info(self, scalar_converter: Callable[[str], str]) -> TypeInfo:
         if self.resolved_pointee_type:
             pointee_type = self.resolved_pointee_type.name
         else:
-            pointee_type = _scalar_python_type(self.pointee_type)
+            pointee_type = scalar_converter(self.pointee_type)
 
         if self.resolved_root_type:
             root_type = self.resolved_root_type.name
         else:
-            root_type = _scalar_python_type(self.root_type)
+            root_type = scalar_converter(self.root_type)
 
-        return f"Reference[{pointee_type}, {root_type}]"
+        return TypeInfo(pointee_type, False, root_type)
```

Comment:
All of this is quite confusing and calls for some re-engineering, but let's go with it for now.